### PR TITLE
Use container log instead of termination log

### PR DESCRIPTION
**Pull request by gjkim42**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Since the termination log cannot be accessed until the container is terminated, use the container log.

#### Which issue(s) this PR fixes:

Fixes #
ref: https://github.com/kubernetes/kubernetes/pull/116398#issuecomment-1700995522

#### Does this PR introduce a user-facing change?

```release-note
none
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/cc @tzneal

### CODE REVIEW

File: test/e2e_node/container_lifecycle_pod_construction.go [hunk 1]
```diff
@@ -250,21 +252,28 @@ func (o containerOutputList) findIndex(name string, command string, startIdx int
...
-			buf.WriteString(cs.State.Terminated.Message)
-		} else if cs.LastTerminationState.Terminated != nil {
-			buf.WriteString(cs.LastTerminationState.Terminated.Message)
+		if cs.State.Running == nil && cs.State.Terminated == nil {
+			continue
 		}
+		log, err := e2epod.GetPodLogs(ctx, f.ClientSet, f.Namespace.Name, pod.Name, cs.Name)
+		framework.ExpectNoError(err)
+		buf.WriteString(log)
+		//		// If the container is terminated but the reason is ContainerStatusUnknown,
```
Comment (tzneal) [inline test/e2e_node/container_lifecycle_pod_construction.go:269]: Remove before merge, assuming you left it in temporarily just to ensure all tests pass.

**Comment by gjkim42** [comment]
/cc @tzneal 
Could you take a look?

/assign @SergeyKanzhelev

**Review by tzneal — APPROVED** [review]
/lgtm

**Comment by gjkim42** [comment]
this is needed to test the https://github.com/kubernetes/kubernetes/pull/120269.

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Comment by gjkim42** [comment]
/hold 
in case there comes new usages for parseLog.
I'll rebase the upstream later and merge it.

**Comment by gjkim42** [comment]
> /hold
> in case there comes new usages for parseLog.
> I'll rebase the upstream later and merge it.

/unhold

rebased the upstream and fixed the new usage.

Could you PTAL? @SergeyKanzhelev @tzneal

**Pull request merged**